__tests__/bin/vip-logs.js

```diff
@@ -76,11 +80,87 @@ describe( 'getLogs', () => {
 			env_id: 3,
 			type: 'app',
 			limit: 500,
+			format: 'console',
 		};
 
 		expect( tracker.trackEvent ).toHaveBeenCalledTimes( 2 );
 		expect( tracker.trackEvent ).toHaveBeenNthCalledWith( 1, 'logs_command_execute', trackingParams );
-		expect( tracker.trackEvent ).toHaveBeenNthCalledWith( 2, 'logs_command_success', trackingParams );
+		expect( tracker.trackEvent ).toHaveBeenNthCalledWith( 2, 'logs_command_success', {
+			...trackingParams,
+			logs_output: 2,
+		} );
+
+		expect( rollbar.error ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should display the logs in the output with JSON format', async () => {
+		opts.format = 'json';
+
+		logsLib.getRecentLogs.mockImplementation( async () => [
+			{ timestamp: '2021-11-05T20:18:36.234041811Z', message: 'My container message 1' },
+			{ timestamp: '2021-11-09T20:47:07.301221112Z', message: 'My container message 2' },
+		] );
+
+		await getLogs( [], opts );
+
+		expect( format.formatData ).toHaveBeenCalledTimes( 1 );
+		expect( format.formatData ).toHaveBeenCalledWith( [
+			{ timestamp: '2021-11-05T20:18:36.234041811Z', message: 'My container message 1' },
+			{ timestamp: '2021-11-09T20:47:07.301221112Z', message: 'My container message 2' },
+		], 'json' );
+
+		const trackingParams = {
+			command: 'vip logs',
+			org_id: 2,
+			app_id: 1,
+			env_id: 3,
+			type: 'app',
+			limit: 500,
+			format: 'json',
+		};
+
+		expect( tracker.trackEvent ).toHaveBeenCalledTimes( 2 );
+		expect( tracker.trackEvent ).toHaveBeenNthCalledWith( 1, 'logs_command_execute', trackingParams );
+		expect( tracker.trackEvent ).toHaveBeenNthCalledWith( 2, 'logs_command_success', {
+			...trackingParams,
+			logs_output: 2,
+		} );
+
+		expect( rollbar.error ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should display the logs in the output with CSV format', async () => {
+		opts.format = 'csv';
+
+		logsLib.getRecentLogs.mockImplementation( async () => [
+			{ timestamp: '2021-11-05T20:18:36.234041811Z', message: 'My container message 1' },
```

Comment:
It would be great to test this with some double quotes and newlines and such
```suggestion
			{ timestamp: '2021-11-05T20:18:36.234041811Z', message: 'My container message "double quotes" and \'single quotes\' and multiple\nlines\nand	tabs' },
```

Reply:
Arguably, this should be left to the CSV formatting lib, but I guess it doesn't hurt to test here. I did change this to spy on the console output instead which is a better test.

89e802d